### PR TITLE
Hopefully fixes consistency issue

### PR DESCRIPTION
**Pull request by WorldofKerry**
- More dynamic recursion limit in pytest.

### CODE REVIEW

File: python2verilog/simulation/iverilog.py [hunk 1]
```diff
@@ -66,17 +67,19 @@ def _run_cmd_with_fifos(
 
         try:
             logging.debug("Waiting on process for %ss", timeout)
+            start_time = time.time()
             process.wait(timeout=timeout)
+            logging.debug("Took %ss", time.time() - start_time)
```
Comment (Haxrox) [inline python2verilog/simulation/iverilog.py:72]: Is this a hack to wait timeout seconds or smth?

File: python2verilog/api/context.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def context_to_codegen(context: ir.Context):
     context.validate()
     ir_root, context = GeneratorFunc(context).create_root()
     logging.debug(
-        "context to codegen %s %s -O%s with %s",
+        "context to codegen %s %s -O%s %s",
```
Comment (Haxrox) [inline python2verilog/api/context.py]: Why remove `with`?

File: python2verilog/simulation/iverilog.py (hunk 1 above)
Comment (WorldofKerry) [reply python2verilog/simulation/iverilog.py:72]: No. It's for letting a debugger know how long the process actually takes. So, if they want to make the timeout proportional to a value, they can more easily find a formula.

File: python2verilog/api/context.py (hunk 2 above)
Comment (WorldofKerry) [reply python2verilog/api/context.py]: Reverted.

**Comment by WorldofKerry** [comment]
I've also modified some recursion limits.

**Review by Haxrox — APPROVED** [review]

**Pull request merged**